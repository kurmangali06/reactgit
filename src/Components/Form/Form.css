button {
  background: green;
  padding: 20px 75px;
  border: 0;
  border-radius: 20px;
  font-size: 16px;
  color: #f2f2f2; 
  display: block;
}

 button:hover {
  transform: scale(1.1);
  transition: 1s ease-in-out;
}
 button:active {
  background: #f2f2f2;
  color: green;
  transform: scale(0.9);
  transition: 1s ease-in-out;
}