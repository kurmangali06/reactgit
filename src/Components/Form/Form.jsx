import React, { useState } from 'react';
import { Button } from './Components/Button/Button';

export const Form = ({ addMessage }) => {
  const [value, setValue] = useState('');

  const handleSubmitForm = (e) => {
    e.preventDefault();
    addMessage(value);
    setValue('');
  };

  return (
    <>
      <form className=" form" onSubmit={handleSubmitForm}>
        <input
          type="text"
          value={value}
          onChange={(ev) => setValue(ev.target.value)}
          className="input"
        />
        <Button disabled={!value} />
      </form>
    </>
  );
};
export default Form;
