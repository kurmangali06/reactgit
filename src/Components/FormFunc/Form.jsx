import React, { useState } from 'react';
import { Button } from "./Button";
import { Input } from "./Input";



export const Form = () => {
  const [name, setName] = useState('click')
  const [value, setValue] = useState('')
  const [messages, setMessages] = useState([])
  const [visible, setVisible] = useState(true)

  const handleClick = () => {
    setMessages([...messages, value])
    setValue('')
  }

  const handleChange = (ev) => {
    setValue(ev.target.value)
  }


  return <>
    <div className='container form'>
      {visible && <ul className='list1a'>
        {messages.map((message, idx) =>
          <li key={idx}>{message}</li>
        )}
      </ul>}

      <Input change={handleChange} value={value} />
      <Button name={name} click={handleClick} />

      <br />
      <button onClick={() => setVisible(!visible)} className='submit' >
        {visible ? 'показать' : 'очистить'}
      </button>
    </div>
  </>
}
export default Form;