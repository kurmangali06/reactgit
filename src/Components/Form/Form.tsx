import React, { FC, useState } from 'react';

import Input from '@mui/material/Input'
import { useDispatch } from 'react-redux';
import { addMessageWithReplay } from '../../store/chats/action';
import { useParams } from 'react-router-dom';
import { AUTHOR } from '../../constants';
import "./Form.css"



export const Form: FC = () => {
  const [value, setValue] = useState('');
  const { chatId } = useParams();

  const dispatch = useDispatch();

  const handleSubmitForm = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (chatId) {
      dispatch(addMessageWithReplay(chatId, { text: value, author: AUTHOR.USER }));

    }

    setValue('');
  };

  return (
    <form className="form" onSubmit={handleSubmitForm}>
      <Input
        placeholder="Введите текст"
        type="text"
        value={value}
        onChange={(ev) => setValue(ev.target.value)}
        className="input"
        data-testid="unput-element"
      />
      <button disabled={!value} >отправить</button>
    </form>
  );
};
export default Form;
