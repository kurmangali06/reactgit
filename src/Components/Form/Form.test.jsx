import React from 'react';
import Form from './Form';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Form', () => {
  it('renders Form component', () => {
    render(<Form />);
  });

  it('have to the input', () => {
    render(<Form />);
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });

  it('render to by placeholder', () => {
    render(<Form />);
    expect(screen.getByPlaceholderText('Введите текст')).toBeInTheDocument();
  });

  it('value is empty string', () => {
    render(<Form />);
    expect(screen.getByDisplayValue('')).toBeInTheDocument();
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Form />);
    expect(asFragment()).toMatchSnapshot();
  });
});
