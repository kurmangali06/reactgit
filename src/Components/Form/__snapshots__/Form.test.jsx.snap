// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form render with snapshot 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <input
      class="input"
      data-testid="unput-element"
      placeholder="Введите текст"
      type="text"
      value=""
    />
    <button
      class="submit"
      disabled=""
      type="submit"
    >
      click
    </button>
  </form>
</DocumentFragment>
`;
